### DISCUSSION TITLE
Termux sponsor using BTT crypto

### DISCUSSION THREAD

**Discussion by simbadMarino** [termux/termux-app#2715] [General]
Hello termux team, I'd like to start being a sponsor of termux development, however can't do it trough traditional ways suggested in the repo for several reasons.

My proposal is for you to create a BTTC or TRON shared wallet between the active development team and publish it on the README. I'll be forever grateful for all the support I've received from you guys so I'm giving back part of the income from dCloud to you guys. For sure this is a win-win situation, let's grow together!

I'm also willing to test hosting a termux-packages mirror using the BitTorrent FileSystem(BTFS) which is what I'm using on [dCloud](https://github.com/simbadMarino/dCloud) and willing to pay for the decentralized storage fees. 

Big thank you ALL!
@agnostic-apollo I made a promise a while back ... ;)

**Comment by agnostic-apollo** [comment]
Check https://github.com/termux/termux-packages/wiki/Donate

You can send the money to @Grimler91 or @thunder-coding crypto addresses. Although, you previously said that ETH fees are high for you. If its for @termux org instead of a specific person, you can notify us so that it can be spent for it. If "some" member doesn't give @termux org the money after receiving, rest assured, the other members will send goons after them to get every penny back!

There are plans for a public ledger for this stuff and other donation sites. Will look into other crypto addresses as well.

>Big thank you ALL!
> @agnostic-apollo I made a promise a while back ... ;)

Yes, yes, I remember all the instances. I will send the "welcome" on receival of the big bucks, I promise ;)

**Reply by simbadMarino** [reply]
> Check https://github.com/termux/termux-packages/wiki/Donate
> 
> You can send the money to @Grimler91 or @thunder-coding crypto addresses. Although, you previously said that ETH fees are high for you. If its for @termux org instead of a specific person, you can notify us so that it can be spent for it. If "some" member doesn't give @termux org the money after receiving, rest assured, the other members will send goons after them to get every penny back!
> 
> There are plans for a public ledger for this stuff and other donation sites. Will look into other crypto addresses as well.
> 
> > Big thank you ALL!
> > @agnostic-apollo I made a promise a while back ... ;)
> 
> Yes, yes, I remember all the instances. I will send the "welcome" on receival of the big bucks, I promise ;)

BTC sent to @thunder-coding (0.003631 BTC). 

I can provide another $800 in TRX OR/AND BTT crypto with the solely condition of adding a TRX AND/OR BTTC QR wallet address(as described below) for donations in the termux README.

**Examples of QR in the new crypto donations section of Termux Readme:**
 
<img width="295" alt="image" src="https://user-images.githubusercontent.com/11146636/164111978-e28d713a-560d-4b0a-9cc0-6a74db5a2c52.png">
 Donate using TRON (TRX)


<img width="295" alt="image" src="https://user-images.githubusercontent.com/11146636/164111447-1fb6e3ea-e5a4-424a-982d-3d5901214e01.png">
Donate using BTT (BitTorrent Chain-BTTC)


**Why TRON / BTTC?**
Yes, TRON /BTTC is cheap to transact, fast and reliable. Crypto is the Internet money and much easier to use than PayPal and other funding mechanism. Also dCloud app revenue will come through  BitTorrent Chain, so its gonna be a lot easier for me to send even small tips for Termux using either TRON or BitTorrent Chain, we won't suffer on BTC or ETH high fees.

**What do I expect from you with these tips?**
Just keep doing what you do with termux!! the extraordinary support I've received from the team is nuts.
As for the use of the $ please use it on yourselves ;) (unless there is a super critical payment for the project you need to do)

Once you have your Wallet Addresses you can create your QRs here: https://qr.io , or any other similar sites

As soon as you have the QR / QRs ready just tag me here to proceed with the transaction.

Thank you everyone contributing on this awesome FOSS project :)

**Reply by agnostic-apollo** [reply]
>BTC sent to @thunder-coding (0.003631 BTC).

Wow, that is indeed BIG bucks! Thanks a lot mate! ❤️

And "You are welcome", as owed ;)

>I can provide another $800 in TRX OR/AND BTT crypto with the solely condition of adding a TRX AND/OR BTTC QR wallet address(as described below) for donations in the termux README.

>As soon as you have the QR / QRs ready just tag me here to proceed with the transaction.

That can be done. I'll discuss with the team about how to manage that, will need to decide which wallet to use and who will manage it, considering its not just about receiving but actually being able to use the money, either to send to members/contributors or to pay for termux related expenses.

>Just keep doing what you do with termux!!

Ah, you mean free labour for years! I see I see, that's the real hidden trap! :p

As a thank you, I'll try to prioritize merging the `updated-flavors` branch, once thunder has confirmed receival of course :p

You are only using a single APK, right, with no other package APKs? Also `build-boostrap.sh` is officially supported now, check https://github.com/termux/termux-packages/wiki/For-maintainers#build-bootstrap-archives

>Thank you everyone contributing on this awesome FOSS project :)

You are very welcome, from me :)

**Reply by thunder-coding** [reply]
> BTC sent to @thunder-coding (0.003631 BTC).

I confirm the recieval of 0.00362406 BTC.

As an open source contributor, this is my first income and I am very happy to see that my work is helping people in the community out there. Thanks for the love ❤️!

**Reply by simbadMarino** [reply]
You are very welcome! Indeed is helping a lot of people out there including me :) 
Thanks and best regards!

**Reply by simbadMarino** [reply]
> > BTC sent to @thunder-coding (0.003631 BTC).
> 
> Wow, that is indeed BIG bucks! Thanks a lot mate! ❤️
> 
> And "You are welcome", as owed ;)
> 
> > I can provide another $800 in TRX OR/AND BTT crypto with the solely condition of adding a TRX AND/OR BTTC QR wallet address(as described below) for donations in the termux README.
> 
> > As soon as you have the QR / QRs ready just tag me here to proceed with the transaction.
> 
> That can be done. I'll discuss with the team about how to manage that, will need to decide which wallet to use and who will manage it, considering its not just about receiving but actually being able to use the money, either to send to members/contributors or to pay for termux related expenses.
> 

I recommend you to do some investigation on wallets multisignature, you can basically control some features like sending funds, etc ONLY after all team members agree on withdrawals, etc. I normally use Tronlink app which has this feature available and its pretty convienent when you need to share wallet .
Some documentation:
https://coredevs.medium.com/tron-multi-signature-mechanism-92ac998993ac
https://tronscanorg.zendesk.com/hc/en-us/articles/360044172251-How-to-use-TRONSCAN-multi-signature-feature 

> > Just keep doing what you do with termux!!
> 
> Ah, you mean free labour for years! I see I see, that's the real hidden trap! :p

![its-a-trap-ackbar](https://user-images.githubusercontent.com/11146636/164267658-c026b8a0-3a09-47d2-9add-4707b375c386.gif)
HAHAHAA, naah, well, maybe, who knows... at the end that's our fate as FOSS developers no? hehe



> As a thank you, I'll try to prioritize merging the `updated-flavors` branch, once thunder has confirmed receival of course :p
> 
Wow thats great!, been using the branch and so far so good !

> You are only using a single APK, right, with no other package APKs? Also `build-boostrap.sh` is officially supported now, check https://github.com/termux/termux-packages/wiki/For-maintainers#build-bootstrap-archives
> 
Yes Im integrating the UI(react native) with termux code into a single APK

> > Thank you everyone contributing on this awesome FOSS project :)
> 
> You are very welcome, from me :)

**Reply by agnostic-apollo** [reply]
> solely condition of adding a TRX AND/OR BTTC QR wallet address(as described below) for donations in the termux README.

Hey, @simbadMarino, well, nobody got around to creating a wallet yet, however, I made a open collective account for termux and I have just enabled crypto donations in it. It does support quite a few currencies, including BTC, but unfortunately not TRX, at least currently. They also automatically convert crypto to fiat and add it to our account, which makes its actually usable for the devs. Maybe, this works for you for now.

https://opencollective.com/termux/donate/crypto

https://opencollective.com/termux

https://github.com/termux/termux-packages/wiki/Donate

https://blog.opencollective.com/support-open-source-software-using-crypto

> HAHAHAA, naah, well, maybe, who knows... at the end that's our fate as FOSS developers no? hehe

I see, I see, I guess soon it will have to be only S developer then! 😈

>Wow thats great!, been using the branch and so far so good !

Great, I plan on getting initial support done before next release, and apollo is the greatest of "planners" :p 

Bootstrap scripts also need updates. Check `TAPM` in b950efec for "how" it will be supported.

>Yes Im integrating the UI(react native) with termux code into a single APK

I see, that would be easier. Multi APK package system has issues, although I do have an idea on how to solve them.

**Reply by simbadMarino** [reply]
Hello @agnostic-apollo hope you are doing great! 
Regarding the crypto payment gateway unfortunately it is part of the deal to include either TRON or BTT :/, I'd even go for XMR(Monero) but most coins supported in the opencollective are Ethereum tokens which are super expensive to transact with :( and not suitable for micro payments in the future... hopefully you can enable a TRON  multi-sign wallet or even a payment gateway accepting TRON/BTT (which I won't suggest as they usually require some fees hehe). I'm available for you to help  on creating the multi-signature wallet for Termux, even trough a  Zoom call if needed. Don't worry, my offer does not expire (unless I die) so no need to rush hehe.

I'm still struggling with the package name but as I focused a little bit on iOS suppor the last month I gave it less priority to find out what was the problem... will work on it later this month hehe.

Best regards!

**Reply by agnostic-apollo** [reply]
There seem to be a few recommended wallets available, but issue is for the wallet crypto to be convertible to usable fiat so that it can be used for paying termux costs since crypto is not accepted everywhere. Crypto may be fine for some stuff though. Crypto or conversion is also banned in some countries, that's an issue too. I guess will keep wallet creation on the todo list, lot of things on the list these days.

>unless I die

Wait, you don't plan to die! Immortality is probably overrated. ;)

>I'm still struggling with the package name 

Didn't understand you.

**Reply by simbadMarino** [reply]
Not planning to die... but know for a fact that I can die tomorrow  lol, regarding the package name I'm still unable to change it to com.justshare instead of com.termux in some parts, need to go really deep to understand what's causing the problem hehe..

I totally get the crypto problem for some of termux members, maybe I'm short sighted cuz my country does not ban it...

I'll check the open collective option, so is this 100% ready for termux now? can I make a deposit and you will receive it with 0 issues?

**Reply by agnostic-apollo** [reply]
>Not planning to die... but know for a fact that I can die tomorrow lol

That can be arranged. Just send some crypto so that I can pay it to certain someone on the dark web! :p

**Reply by agnostic-apollo** [reply]
>'m still unable to change it to com.justshare instead of com.termux in some parts, need to go really deep to understand what's causing the problem hehe..

Can you tell me which part. I do plan on working on it in the next few days probably.

>my country does not ban it...

Lucky you. Well, other than all the crypto crashes, yikes! 

>I'll check the open collective option, so is this 100% ready for termux now? can I make a deposit and you will receive it with 0 issues?

Some people have already sent fiat and we have received it in our account, but not crypto, so you will be the first. ;) It should work since we are approved and all. We haven't extracted anything yet since not much received, but we already discussed alternative payout methods with open collective and hopefully all members should be able to get it, either via paypal, wise or payoneer. If someone wants to receive crypto, they will likely have to go through someone else, or should be donated to directly, since I am pretty sure payouts in crypto won't be possible.

https://docs.opencollective.com/help/fiscal-hosts/payouts

**Reply by simbadMarino** [reply]
Fair enough you convinced me, let me do some trading to the supported cryptos and will donate :)


In regards of the specific package problems I'm facing let me finish some stuff and will properly document my problems, it's kinda difficult to explain in a few words, I'll need to point to the code lines and make some notes as well, right now a bit lazy to do it hehe but I feel its because my main package name is com.justshare and I'm using the com.termux package as it is, maybe I'll need to change the termux package name and refactor lots of stuff hehe.

**Reply by agnostic-apollo** [reply]
Awesome. Thanks :)

Sure. Take your time. 

>I'm using the com.termux package as it is

If you don't move termux classes to `com.justshare` package and keep them in `com.termux`, then some intents like in `termux-tools` scripts may break since compiled scripts would have `com.justshare/com.justshare.app.TermuxActivity`, but in your apk the compiled classes would have `com.termux.app.TermuxActivity`.

The build scripts need to support an additional variable in addition to `TERMUX_PACKAGE_MAME` to support cases where app package name does not equal to classes package name.

https://github.com/termux/termux-app/blob/231ecff5f0837d1aa2831039b64afd07915e1cab/termux-shared/src/main/java/com/termux/shared/termux/TermuxConstants.java#L280

**Reply by agnostic-apollo** [reply]
[Damn @simbadMarino, JUST DAMN!](https://opencollective.com/dcloud). Thanks a lot dude! Really really appreciate it! ❤️

**Reply by simbadMarino** [reply]
You are very welcome, thanks to you guys! I hope we can keep collaborating in the near future, I have some cool ideas we could use to reduce your storage expenses. Also willing to donate more as soon as the crypto market recovers a little lol.

**Reply by agnostic-apollo** [reply]
Yeah, sure we can. lolz. good luck on that!
